keras_nlp/src/utils/preset_utils.py

```diff
@@ -704,35 +705,57 @@ def load_tokenizer(self, cls, **kwargs):
         tokenizer.load_preset_assets(self.preset)
         return tokenizer
 
-    def load_task(self, cls, load_weights, **kwargs):
+    def load_task(self, cls, load_weights, load_task_extras, **kwargs):
         # If there is no `task.json` or it's for the wrong class delegate to the
         # super class loader.
+        if not load_task_extras:
+            return super().load_task(
+                cls, load_weights, load_task_extras, **kwargs
+            )
         if not check_file_exists(self.preset, TASK_CONFIG_FILE):
-            return super().load_task(cls, load_weights, **kwargs)
+            raise ValueError(
+                "Saved preset has no `task.json`, cannot load the task config "
+                "from a file. Call `from_preset()` with "
+                "`load_task_extras=False` to load the task from a backbone "
+                "with library defaults."
+            )
         task_config = load_json(self.preset, TASK_CONFIG_FILE)
         if not issubclass(check_config_class(task_config), cls):
-            return super().load_task(cls, load_weights, **kwargs)
+            raise ValueError(
+                f"Saved `task.json`does not match calling cls {cls}. Call "
+                "`from_preset()` with `load_task_extras=False` to load the "
+                "task from a backbone with library defaults."
+            )
         # We found a `task.json` with a complete config for our class.
         task = load_serialized_object(task_config, **kwargs)
         if task.preprocessor is not None:
             task.preprocessor.tokenizer.load_preset_assets(self.preset)
         if load_weights:
-            jax_memory_cleanup(task)
+            jax_memory_cleanup(task.backbone)
```

Comment:
Don't we need a `jax_memory_cleanup` to clean up head weights when `load_task_extras` is `True`?

Reply:
SG! I can push this clean up inside an if else block. Practically it won't matter today, as all our task weights are tiny compared to backbones, but that might not always be the case for all tasks.